doc/Security.xml

```diff
@@ -1091,10 +1091,16 @@
     <section xml:id="section_b4m_rs4_rwb">
       <title>Authorization Servers</title>
       <para>This section describes use of external authorization servers that can be used for
-        providing authentication of users and authorizing access to both device services and
-        external services, like cloud storage.</para>
+        providing authentication of users and authorizing access to both internal device services
+        and external services, like cloud storage.</para>
       <para>OAuth 2.0 is a specification that describes how to issue access tokens. It is defined in RFC6749. OpenID Connect has been developed as an extension of OAuth 2.0.</para>
-      <para>OAuth 2.0 includes the definition of a web API called "authorization endpoint". Requests to this API must include a parameter called <emphasis>response_type</emphasis> which can be either <emphasis>code</emphasis> or <emphasis>token</emphasis>. OpenID Connect extends the OAuth 2.0 specification, by adding the values <emphasis>id_token</emphasis> and <emphasis>none</emphasis> for <emphasis>response_type</emphasis> and allowing any combination of <emphasis>code</emphasis>, <emphasis>token</emphasis> and <emphasis>id_token</emphasis>.</para>
+      <para>OAuth 2.0 includes the definition of a web API called "authorization endpoint". Requests
+        to this API must include a parameter called <emphasis>response_type</emphasis> with value as
+          <emphasis>code</emphasis> for authorization code flow OR parameter called
+          <emphasis>grant_type</emphasis> with value as <emphasis>client_credentials</emphasis> for
```

Comment:
Rereading this sentence, we are talking specifically about the "authorization endpoint" here. Which is only used for the user authorization flow.
In the machine authentication flow, we never talk to that endpoint, we only interact with the token endpoint ([RFC 6749 Section 3.2](https://datatracker.ietf.org/doc/html/rfc6749#section-3.2)) so I'm not sure the wording is correct here.
I'm a bit confused that we felt the need to mention this section, but didn't mention the token endpoint at all.
I would split the sentence here, stopping at "for authorization code flow."
And maybe add a distinct paragraph for the "token endpoint", mentioning the use of either "grant_type=authorization_code" for the user authentication or "grant_type=client_credentials" for the device authentication?

Reply:
I agree with your comment, As we discussed in the telco we need to check how we can organize the spec to avoid these confusions and may be wherever possible just refer to RFC sections. Any other suggestions are welcome.